cvss2/base_metrics.go

```diff
@@ -0,0 +1,230 @@
+	return weightsAccessComplexity[ac]
+}
+
+func (ac AccessComplexity) String() string {
+	return codeAccessComplexity[ac]
+}
+
+func (ac *AccessComplexity) parse(str string) error {
+	idx, found := findIndex(str, codeAccessComplexity)
+	if found {
+		*ac = AccessComplexity(idx)
+		return nil
+	}
+	return fmt.Errorf("illegal access complexity code %s", str)
+}
+
+type Authentication int
+
+const (
+	AuthenticationMultiple Authentication = iota + 1
+	AuthenticationSingle
+	AuthenticationNone
+)
+
+var (
+	weightsAuthentication = []float64{0, 0.45, 0.56, 0.704}
+	codeAuthentication    = []string{"", "M", "S", "N"}
+)
+
+func (au Authentication) defined() bool {
+	return int(au) != 0
+}
+
+func (au Authentication) weight() float64 {
+	return weightsAuthentication[au]
+}
+
+func (au Authentication) String() string {
+	return codeAuthentication[au]
+}
+
+func (au *Authentication) parse(str string) error {
+	idx, found := findIndex(str, codeAuthentication)
+	if found {
+		*au = Authentication(idx)
+		return nil
+	}
+	return fmt.Errorf("illegal authentication code %s", str)
+}
+
+type ConfidentialityImpact int
+
+const (
+	ConfidentialityImpactNone ConfidentialityImpact = iota + 1
+	ConfidentialityImpactPartial
+	ConfidentialityImpactComplete
+)
+
+var (
+	weightsConfidentialityImpact = []float64{0, 0, 0.275, 0.66}
+	codeConfidentialityImpact    = []string{"", "N", "P", "C"}
+)
+
+func (ci ConfidentialityImpact) defined() bool {
+	return int(ci) != 0
+}
+
+func (ci ConfidentialityImpact) weight() float64 {
+	return weightsConfidentialityImpact[ci]
+}
+
+func (ci ConfidentialityImpact) String() string {
+	return codeConfidentialityImpact[ci]
+}
+
+func (ci *ConfidentialityImpact) parse(str string) error {
+	idx, found := findIndex(str, codeConfidentialityImpact)
+	if found {
+		*ci = ConfidentialityImpact(idx)
+		return nil
+	}
+	return fmt.Errorf("illegal confidentiality impact %s", str)
+}
+
+type IntegrityImpact int
+
+const (
+	IntegerityImpactNone IntegrityImpact = iota + 1
+	IntegrityImpactPartial
+	IntegrityImpactComplete
+)
+
+var (
+	weightsIntegrityImpact = []float64{0, 0, 0.275, 0.66}
+	codeIntegrityImpact    = []string{"", "N", "P", "C"}
+)
+
+func (ii IntegrityImpact) defined() bool {
+	return int(ii) != 0
+}
+
+func (ii IntegrityImpact) weight() float64 {
+	return weightsIntegrityImpact[ii]
+}
+
+func (ii IntegrityImpact) String() string {
+	return codeIntegrityImpact[ii]
+}
+
+func (ii *IntegrityImpact) parse(str string) error {
+	idx, found := findIndex(str, codeIntegrityImpact)
+	if found {
+		*ii = IntegrityImpact(idx)
+		return nil
+	}
+	return fmt.Errorf("illegal integrity impact code %s", str)
+}
+
+type AvailabilityImpact int
+
+const (
+	AvailabilityImpactNone AvailabilityImpact = iota + 1
+	AvailabilityImpactPartial
+	AvailabilityImpactComplete
+)
+
+var (
+	weightsAvailabilityImpact = []float64{0, 0, 0.275, 0.66}
+	codeAvailabilityImpact    = []string{"", "N", "P", "C"}
+)
+
+func (ai AvailabilityImpact) defined() bool {
```

Comment:
Just heads up, not  a request for action, but it is idiomatic to keep receivers consistent, unless some other considerations were specifically taken: https://golang.org/doc/faq#methods_on_values_or_pointers
```
Next is consistency. If some of the methods of the type must have pointer receivers, the rest should too, so the method set is consistent regardless of how the type is used. See the section on method sets for details. 
```

Reply:
I won't change this for now. Maybe in the future for both cvss 2 and 3